src/main/java/org/opentripplanner/middleware/models/MobilityProfile.java

```diff
@@ -0,0 +1,116 @@
+     * @param mobilityProfile consulted to construct and update mobility mode
+     */
+    public static void updateMobilityMode(MobilityProfile mobilityProfile) {
+        // Variable names and the strings we parse are from Georgia Tech document, to facilitate syncing
+        // changes.  The testing for devices and vision in this order are from the same document; note
+        // that this means the devices tested for later will override the earlier "Temp"orary settings.
+        String mModeTemp = "None";
+        String visionTemp = "None";
+
+        // If "none" has been specified at all, we just wipe the mobility devices clear,
+        // else we look at the mobility devices and settle on the one that is the most involved.
+        if (mobilityProfile.mobilityDevices.contains("none")) {
+            mobilityProfile.mobilityDevices.clear();
+        } else {
+            if (mobilityProfile.mobilityDevices.contains("white cane")) {
+                visionTemp = "Blind";
+            }
+            if (mobilityProfile.mobilityDevices.contains("manual walker")
+                    || mobilityProfile.mobilityDevices.contains("wheeled walker")
+                    || mobilityProfile.mobilityDevices.contains("cane")
+                    || mobilityProfile.mobilityDevices.contains("crutches")
+                    || mobilityProfile.mobilityDevices.contains("stroller")
+                    || mobilityProfile.mobilityDevices.contains("service animal")) {
+                mModeTemp = "Device";
+            }
+            if (mobilityProfile.mobilityDevices.contains("mobility scooter")) {
+                mModeTemp = "MScooter";
+            }
+            if (mobilityProfile.mobilityDevices.contains("electric wheelchair")) {
+                mModeTemp = "WChairE";
+            }
+            if (mobilityProfile.mobilityDevices.contains("manual wheelchair")) {
+                mModeTemp = "WChairM";
+            }
+
+            if ("None".equals(mModeTemp) && mobilityProfile.isMobilityLimited) {
+                mModeTemp = "Some";
+            }
+        }
+
+        if (MobilityProfile.VisionLimitation.LOW_VISION == mobilityProfile.visionLimitation) {
+            visionTemp = "LowVision";
+        } else if (MobilityProfile.VisionLimitation.LEGALLY_BLIND == mobilityProfile.visionLimitation) {
+            visionTemp = "Blind";
+        }
+
+        // Create combinations for mobility mode and vision
+        if (Set.of("LowVision", "Blind").contains(visionTemp)) {
+            if ("None".equals(mModeTemp)) {
+                mobilityProfile.mobilityMode = visionTemp;
+            } else if (MOBILITY_DEVICES.contains(mModeTemp)) {
+                mobilityProfile.mobilityMode = mModeTemp + "-" + visionTemp;
+            }
+        } else if (MOBILITY_DEVICES.contains(mModeTemp)) {
+            mobilityProfile.mobilityMode = mModeTemp;
+        }
+        mobilityProfile.mobilityMode = "None";
```

Comment:
I think this is a refactor mistake. This will override any previous computations of `mobilityMode`...

Reply:
Yikes, definitely a rogue edit. Will fix.